dbt_subprojects/solana/models/_sector/dex/phantom_swapper/phantom_swapper_solana_trades.sql

```diff
@@ -0,0 +1,105 @@
+{{ config(
+    schema = 'phantom_swapper_solana',
+    alias = 'bot_trades',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    incremental_strategy = 'merge',
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+    unique_key = ['block_month', 'block_slot', 'tx_index', 'outer_instruction_index', 'inner_instruction_index']
+   )
+}}
+
+{% set query_start_date = '2025-04-01' %}
+{% set wsol_token = 'So11111111111111111111111111111111111111112' %}
+
+with filtered_transactions as (
+        select 
+            id, 
+            signer,
+            block_date
+        from {{ source('solana', 'transactions') }}
+        where
+            {% if is_incremental() %} 
+                {{ incremental_predicate('block_date') }}
+            {% else %} 
+                block_date >= timestamp '{{query_start_date}}'
+            {% endif %} 
+    ),
+    trades as (
+        select
+            'solana' as blockchain,
+            trades.block_time,
+            cast(date_trunc('day', trades.block_time) as date) as block_date,
+            cast(date_trunc('month', trades.block_time) as date) as block_month,
+            trades.amount_usd,
+            if(token_sold_mint_address = '{{wsol_token}}', 'Buy', 'Sell') as type,
+            token_bought_amount,
+            token_bought_symbol,
+            token_bought_mint_address as token_bought_address,
+            token_sold_amount,
+            token_sold_symbol,
+            token_sold_mint_address as token_sold_address,
+            fee_payments.amount_usd as fee_usd,
+            fee_payments.amount as fee_token_amount,
+            fee_payments.token_symbol as fee_token_symbol,
+            fee_payments.token_address as fee_token_address,
+            project,
+            trades.version,
+            token_pair,
+            project_program_id as project_contract_address,
+            trader_id as user,
+            trades.tx_id,
+            trades.block_slot,
+            tx_index,
+            outer_instruction_index,
+            inner_instruction_index
+        from {{ ref('dex_solana_trades') }} as trades
```

Comment:
```suggestion
        from {{ source('dex_solana', 'trades') }} as trades
```

Reply:
until we can resolve run times on all bot trades, we have updated to read as a source to break off the lineage. we will look to tie back together with ref usage when we're ready.

Reply:
Cool. I've set the normal start dates and changed solana.trades reference from ref to source now. 👍